.github/workflows/codeql-analysis.yml

```diff
@@ -266,3 +266,14 @@ jobs:
             echo "::error::Rec clang-tidy failed"
             exit 1
           fi
+
+  check-for-binaries:
+    runs-on: ubuntu-22.04
+    name: Force failure in case there are binaries present in a pull request
+    if: ${{ github.event_name == 'pull_request' }}
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 2
```

Comment:
This scares me...

Reply:
why?

Reply:
fetch-depth 2 in theory only gives you two commits of history, if i make a PR series with 10 commits, then I don't expect you to be able to get good answers from git when you ask it about that.

Reply:
It was a little bit confusing for me too. 

For a pull request, the checkout action creates a "merge commit" which aggregates all commits in the PR, where `HEAD` will be pointing to (unless we specify otherwise, see [here](https://github.com/actions/checkout#Checkout-pull-request-HEAD-commit-instead-of-merge-commit) and [here](https://github.com/actions/checkout/issues/426)). Then we compare this commit with its parent, `HEAD^`.

It is the same syntax we are already using in the `analyze` job of the `codeql` workflow, [line 64](https://github.com/PowerDNS/pdns/blob/master/.github/workflows/codeql-analysis.yml#L64) to later do a similar `git diff` command for `clang-tidy`, in [line 128](https://github.com/PowerDNS/pdns/blob/master/.github/workflows/codeql-analysis.yml#L128).